### DISCUSSION TITLE
Data type `float16` not recognized

### DISCUSSION THREAD

**Discussion by marcdelabarrera** [duckdb/duckdb#5922] [Ideas]
`duckdb` can't create a table from a pandas dataframe if it contains a `float16` column. It would be nice that `float16` were parsed to `float32` and avoid this error.

**Comment by Mytherin** [comment]
Could you provide a code example?

**Comment by marcdelabarrera** [comment]
Sure, this would raise the error:
```
import pandas as pd
import duckdb

df = pd.DataFrame({'col': [1,2,3]})
df = df.astype({'col':'float16'})
con = duckdb.connect()
con.execute('CREATE TABLE tbl AS SELECT * FROM df')
```

```
Traceback (most recent call last):
  File "c:\Users\MBBar\Downloads\example_duckdb.py", line 7, in <module>
    con.execute('CREATE TABLE tbl AS SELECT * FROM df')
duckdb.NotImplementedException: Not implemented Error: Data type 'float16' not recognized
```
Changing the type to `float32` then it does not. In those cases, it would be nice that `duckdb` identified this issue and converted all `float16` columns into `float32` columns.
